### ISSUE TITLE
Question: How to properly resume a download after a network interruption?

### ISSUE THREAD

**Issue by fiso64** [jpdillingham/Soulseek.NET#823]
I've been trying to figure out how to properly resume an incomplete download after a network interruption. 

I'm using the following code to try downloading a file, and retry after a network interruption:
```csharp
int maxRetries = 3;
int retryCount = 0;
while (true)
{
    try
    {
        await client.DownloadAsync(response.Username, file.Filename,
            () => Task.FromResult((Stream)outputStream),
            file.Size, startOffset: outputStream.Position,
            options: transferOptions, cancellationToken: cts.Token);

        break;
    }
    catch (SoulseekClientException ex)
    {
        retryCount++;

        if (IsConnectedAndLoggedIn() || retryCount >= maxRetries)
            throw;

        await WaitForInternetAndLogin(); 
        await Task.Delay(TimeSpan.FromSeconds(1));
    }
}
```

To test it, I download a file, and while it's downloading I disconnect my computer from the internet, then reconnect. The problem is that whenever it tries to resume, the download is always queued remotely and never continues. On my server with slskd I can see that it logs the following message when that happens:
```
Upload of /path/to/file to username is already queued or in progress
```

Interestingly, I've been able to successfully resume from some other users, although it does not work consistently, possibly due to different clients.

**Comment by jpdillingham** [comment]
It might take the other end a while to realize that the connection has been closed.  Give it a bit more time before re-requesting and see if that helps.

Under the hood the process for resuming a transfer is exactly the same as initiating one, the only difference is the starting offset.

**Comment by fiso64** [comment]
I just noticed that I had been (foolishly) testing it while behind a vpn.

Resuming works just fine under normal conditions. I don't even have to add any code myself, it does it all on its own without ever throwing any exceptions from DownloadAsync or me having to reconnect the client (it stays connected and logged in even without internet). For the abnormal disconnect while behind a vpn I tried enabling keepalive with the default settings from slskd, but it didn't help. 

Anyways, the same thing happens if I do this in another client like nicotine. I guess I'll just wait until the remote accepts.

Thanks for the help
